chia/wallet/wallet.py

```diff
@@ -363,13 +362,24 @@ async def _generate_unsigned_transaction(
             )
         elif exclude_coins is not None:
             raise ValueError("Can't exclude coins when also specifically including coins")
+
         assert len(coins) > 0
         self.log.info(f"coins is not None {coins}")
         spend_value = sum([coin.amount for coin in coins])
-
+        self.log.info(f"spend_value is {spend_value} and total_amount is {total_amount}")
         change = spend_value - total_amount
         if negative_change_allowed:
             change = max(0, change)
+        if change < 0 < fee:
+            fee_coins = await self.select_coins(
+                uint64(fee),
```

Comment:
Should this be fee + change (which is sematically the same as fee - abs(change)).  You might be able to partially complete the fee with leftover change, where this requires you to have amount + change + fee in your wallet rather than amount + fee.